src/components/tasks/Monday.tsx

```diff
@@ -279,52 +286,73 @@ const Monday = (props: MondayProps) => {
-        select
-        label="Select Workspace"
-        required
-        value={selectedWorkspace}
-        onChange={(e) => {
-          setSelectedWorkspace(e.target.value);
-        }}
-      >
-        {mondayWorkspacesData?.mondayWorkspaces.elements?.map(
-          (mondayWorkspaceElement) => (
-            <MenuItem
-              key={mondayWorkspaceElement.id}
-              value={mondayWorkspaceElement.id}
-            >
-              {mondayWorkspaceElement.name}
-            </MenuItem>
-          ),
-        ) ?? (
-          <MenuItem disabled>
-            There are no workspace available to select
-          </MenuItem>
+      <Controller
+        render={({ field: { onChange, ...rest } }) => (
+          <TextField
+            select
+            label="Select Workspace"
+            required
+            onChange={(e) => {
+              resetField("element.group", { defaultValue: null });
+              onChange(e.target.value);
+            }}
+            {...rest}
+          >
+            {mondayWorkspacesData?.mondayWorkspaces.elements?.map(
+              (mondayWorkspaceElement) => (
+                <MenuItem
+                  key={mondayWorkspaceElement.id}
+                  value={mondayWorkspaceElement.id}
+                >
+                  {mondayWorkspaceElement.name}
+                </MenuItem>
+              ),
+            ) ?? (
+              <MenuItem disabled>
+                There are no workspace available to select
+              </MenuItem>
+            )}
+          </TextField>
         )}
-      </TextField>
+        name="element.workspace"
+        control={control}
+        defaultValue=""
+      />
       <ClickAwayListener onClickAway={() => setOpenAutocomplete(false)}>
         <Box>
-          <Autocomplete
-            options={[]}
-            renderInput={(params) => (
-              <TextField {...params} label="Select Board > Group" required />
-            )}
-            open={openAutocomplete}
-            onOpen={() => setOpenAutocomplete(true)}
-            value={selectedGroup ?? { id: null, name: null }}
-            getOptionLabel={(o) =>
-              o?.name ? `${o.board?.name} > ${o.name}` : ""
-            }
-            disabled={!selectedWorkspace}
-            openOnFocus
-            disableClearable
-            disableCloseOnSelect
-            PaperComponent={() => (
-              <Paper sx={{ px: 1, py: 2, maxHeight: 200, overflow: "auto" }}>
+          <Controller
+            render={({ field: { onChange, value, ...rest } }) => (
+              <Autocomplete
+                options={[]}
+                renderInput={(params) => (
+                  <TextField
+                    {...params}
+                    label="Select Board > Group"
+                    required
+                  />
+                )}
+                open={openAutocomplete}
+                onOpen={() => setOpenAutocomplete(true)}
+                {...rest}
```

Comment:
To avoid confusion and later error, isn't it better to put rest at the first position ?